core/backend/src/BackendHubAccess.ts

```diff
@@ -227,6 +227,12 @@ export interface BackendHubAccess {
    */
   queryV2Checkpoint: (arg: CheckpointProps) => Promise<V2CheckpointAccessProps | undefined>;
 
+  /**
+   * Get the access props for a successful V2 checkpoint that is at the requested changesetId or the most recent checkpoint before it. Returns undefined if no such V2 checkpoint exists.
+   * @internal
+   */
+  queryCurrentOrPrecedingV2Checkpoint?: (arg: CheckpointProps) => Promise<V2CheckpointAccessProps | undefined>;
```

Comment:
I see no need to make a new method. Please change CheckpointProps to allow the `changeset` be optional (for acquiring new briefcases) and add a new member called `orPreceeding` (or something like that) for snapshots.

Reply:
When the changeset is optional and the user is intending to acquire a new briefcase, is that a briefcase at the latest v2 checkpoint? Or would that be a briefcase at the tip of the imodel? So they'd get the latest v2 checkpoint and then potentially apply changesets?

Reply:
Acquiring a briefcase starts by downloading a checkpoint. That should _always_ be the latest checkpoint. That's what queryV2Checkpoint with `undefined` for changeset would be used for.

Reply:
There are quite a few places (~10) which depend on "changesetId" being defined in the CheckpointProps. I guess if it were undefined, then queryV2Checkpoint would have to return the changesetId that its returning. Technically its part of the dbName in the "V2CheckpointAccessProps" but likely has ".bim" at the end. 

Looking at BriefcaseManager.downloadBriefcase which takes a "RequestNewBriefcaseArg", this already has an optional IModelVersionProps. If IModelVersionProps is not provided, then we use IModelVersion.latest().toJSON().  This results in using a function "getLatestChangeset" (used in the BackendHubAccess implementations) to satisfy the "latest" query.  I would think this makes allowing an undefined changeset unnecessary and saves us the trouble of addressing the many places which don't expect changeset to be undefined. Would you be okay with keeping changeset as required?

Reply:
OK. What happens today if you ask for a checkpoint that doesn't exists?

Reply:
If you ask for a v2 checkpoint that doesn't exist then you'll receive undefined or an error saying that checkpoint does not exist. The addition of allowPreceding to CheckpointProps will resolve this.

In the downloadBriefcase case when you're attempting to get the latest changeset, allowPreceding will make it so we receive the latest v2 checkpoint and apply changesets to that to get to the latest changeset.

Reply:
ok, so the only change we need is a new `allowPreceding` flag on CheckpointProps? 

It is somewhat annoying that you have to do two roundtrips to get the latest checkpoint, but that's not the end of the world.

Reply:
Yes, I believe so.